test/uri/uri_path_test.cc

```diff
@@ -1,64 +1,261 @@
 #include <gtest/gtest.h>
 #include <sourcemeta/jsontoolkit/uri.h>
+#include <string_view>
 
-TEST(URI_path, https_example_url_no_path) {
+// Getter
+
+TEST(URI_path_getter, https_example_url_no_path) {
   const sourcemeta::jsontoolkit::URI uri{"https://example.com"};
   EXPECT_FALSE(uri.path().has_value());
 }
 
-TEST(URI_path, https_example_url_slash) {
+TEST(URI_path_getter, https_example_url_slash) {
   const sourcemeta::jsontoolkit::URI uri{"https://example.com/"};
   EXPECT_TRUE(uri.path().has_value());
   EXPECT_EQ(uri.path().value(), "/");
 }
 
-TEST(URI_path, https_example_url_single) {
+TEST(URI_path_getter, https_example_url_single) {
   const sourcemeta::jsontoolkit::URI uri{"https://example.com/foo"};
   EXPECT_TRUE(uri.path().has_value());
   EXPECT_EQ(uri.path().value(), "/foo");
 }
 
-TEST(URI_path, https_example_url_multi) {
+TEST(URI_path_getter, https_example_url_multi) {
   const sourcemeta::jsontoolkit::URI uri{"https://example.com/foo/bar/baz"};
   EXPECT_TRUE(uri.path().has_value());
   EXPECT_EQ(uri.path().value(), "/foo/bar/baz");
 }
 
-TEST(URI_path, relative_multi) {
+TEST(URI_path_getter, relative_multi) {
   const sourcemeta::jsontoolkit::URI uri{"../foo/bar"};
   EXPECT_TRUE(uri.path().has_value());
   EXPECT_EQ(uri.path().value(), "../foo/bar");
 }
 
-TEST(URI_path, urn) {
+TEST(URI_path_getter, urn) {
   const sourcemeta::jsontoolkit::URI uri{"urn:example:schema"};
   EXPECT_TRUE(uri.path().has_value());
   EXPECT_EQ(uri.path().value(), "example:schema");
 }
 
-TEST(URI_path, urn_with_fragment) {
+TEST(URI_path_getter, urn_with_fragment) {
   const sourcemeta::jsontoolkit::URI uri{"urn:example:schema#foo"};
   EXPECT_TRUE(uri.path().has_value());
   EXPECT_EQ(uri.path().value(), "example:schema");
 }
 
-TEST(URI_path, tag) {
+TEST(URI_path_getter, tag) {
   const sourcemeta::jsontoolkit::URI uri{
       "tag:bowtie.report,2023-11:referencing-suite-tag-uris-id"};
   EXPECT_TRUE(uri.path().has_value());
   EXPECT_EQ(uri.path().value(),
             "bowtie.report,2023-11:referencing-suite-tag-uris-id");
 }
 
-TEST(URI_path, tag_with_fragment) {
+TEST(URI_path_getter, tag_with_fragment) {
   const sourcemeta::jsontoolkit::URI uri{
       "tag:bowtie.report,2023-11:referencing-suite-tag-uris-id#foo"};
   EXPECT_TRUE(uri.path().has_value());
   EXPECT_EQ(uri.path().value(),
             "bowtie.report,2023-11:referencing-suite-tag-uris-id");
 }
 
-TEST(URI_path, without_scheme) {
+TEST(URI_path_getter, without_scheme) {
   const sourcemeta::jsontoolkit::URI uri{"example.com/foo"};
   EXPECT_EQ(uri.path().value(), "example.com/foo");
 }
+
+// Setter
+
+TEST(URI_path_setter, no_path) {
+  sourcemeta::jsontoolkit::URI uri{"https://example.com"};
+
+  uri.path(std::string{"/foo"});
+  EXPECT_EQ(uri.path().value(), "/foo");
```

Comment:
What happens if I set a path that is relative, like `../foo/bar`? You probably want to reject it by throwing an exception, as that case should make use of the URI resolution methods instead

Reply:
hmm let me check!